dask/dataframe/core.py

```diff
@@ -1669,7 +1669,11 @@ def mean(self, axis=None, skipna=True, split_every=False, dtype=None, out=None):
             )
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
+            dtype = getattr(self, "dtype", None)
```

Comment:
This doesn't look quite right. Why was this block of changes necessary? Does `_get_numeric_data` not return what we want?

Reply:
there was a problem when summing.
```
>>> import dask.dataframe as dd
>>> import pandas as pd
>>> dds = dd.from_pandas(pd.Series(pd.date_range('2000', periods=4)), npartitions=1)
>>> num = dds._get_numeric_data()
>>> num.sum()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ec2-user/.local/lib/python3.6/site-packages/dask/dataframe/core.py", line 1522, in sum
    "sum", axis=axis, skipna=skipna, split_every=split_every, out=out
  File "/home/ec2-user/.local/lib/python3.6/site-packages/dask/dataframe/core.py", line 1471, in _reduction_agg
    meta = getattr(self._meta_nonempty, name)(axis=axis, skipna=skipna)
  File "/home/ec2-user/.local/lib/python3.6/site-packages/pandas/core/generic.py", line 11585, in stat_func
    min_count=min_count,
  File "/home/ec2-user/.local/lib/python3.6/site-packages/pandas/core/series.py", line 4102, in _reduce
    **kwds
  File "/home/ec2-user/.local/lib/python3.6/site-packages/pandas/core/base.py", line 1220, in _reduce
    klass=self.__class__.__name__, op=name
TypeError: DatetimeIndex cannot perform the operation sum
```

Reply:
Mmm so the problem is that Dask defines mean in terms of a sum and divide, OK? So we will need some kind of conversion to i8 values. However,

1. We have to be careful with missing values. The i8 conversion will make NaT values the minimum integer, which will throw off the calculation
2. We shouldn't be doing an `astype` on the entire `_get_numeric_data`. What if we have a DataFrame with a mix of dtypes? Rather, we should convert just the datetime values.